pkg/cluster/kubernetes/sync.go

```diff
@@ -208,21 +207,23 @@ func (c *Cluster) getAllowedResourcesBySelector(selector string) (map[string]*ku
 		listOptions.LabelSelector = selector
 	}
 
-	_, resources, err := c.client.discoveryClient.ServerGroupsAndResources()
-	if err != nil {
-		discErr, ok := err.(*discovery.ErrGroupDiscoveryFailed)
-		if !ok {
-			return nil, err
-		}
-		for gv, e := range discErr.Groups {
-			if gv.Group == "metrics" || strings.HasSuffix(gv.Group, "metrics.k8s.io") {
-				// The Metrics API tends to be misconfigured, causing errors.
-				// We just ignore them, since it doesn't make sense to sync metrics anyways.
-				continue
-			}
-			// Tolerate empty GroupVersions due to e.g. misconfigured custom metrics
-			if e.Error() != fmt.Sprintf("Got empty response for: %v", gv) {
-				return nil, err
+	sgs, err := c.client.discoveryClient.ServerGroups()
+	if sgs == nil {
+		return nil, err
+	}
+
+	resources := []*meta_v1.APIResourceList{}
+	for i := range sgs.Groups {
+		gv := sgs.Groups[i].PreferredVersion.GroupVersion
+		// exclude the *.metrics.k8s.io resources to avoid querying the cluster metrics
+		if !strings.Contains(sgs.Groups[i].Name, "metrics.k8s.io") {
```

Comment:
`Contains` seems a little scattershot, can it be more specific? For example, will the name _start_ with metrics.k8s.io; or, if it's a "subdomain", will it _end_ with `.metrics.k8s.io`.

Reply:
Doing two comparisons (starts with and ends with) is faster than contains? The `metics.k8s.io` domain is reserved to Kubernetes aggregator API, I don't see how is a scattershot.

Reply:
> Doing two comparisons (starts with and ends with) is faster than contains?

I'm not suggesting doing two comparisons; I'm suggesting being more precise with the one comparison.

> The `metrics.k8s.io` domain is reserved to Kubernetes aggregator API

If that's the exact name, then test for equality. If the items you want to avoid will be `<something>.metrics.k8s.io`, test that they end with `.metrics.k8s.io`. Otherwise it could exclude some future group "whatever-metrics.k8s.io" inadvertently. Sure, it is not that likely such a group would be introduced, but: better to succeed precisely than to fail accidentally.